src/formatDate.js

```diff
@@ -8,7 +8,43 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const partsDate = date.split(fromFormat[fromFormat.length - 1]);
+  const correctFormat = [...toFormat];
+  const separator = correctFormat[correctFormat.length - 1];
+
+  correctFormat.pop();
+
+  for (let i = 0; i < correctFormat.length; i++) {
+    let value = partsDate[fromFormat.indexOf(correctFormat[i])];
+
+    if (correctFormat[i] === 'DD') {
+      correctFormat[i] = value;
+    }
+
+    if (correctFormat[i] === 'MM') {
+      correctFormat[i] = value;
+    }
+
+    if (
+      (correctFormat[i] === 'YY' || correctFormat[i] === 'YYYY') &&
+      fromFormat.includes(correctFormat[i])
+    ) {
+      correctFormat[i] = value;
+    } else if (correctFormat[i] === 'YY') {
+      value = partsDate[fromFormat.indexOf('YYYY')].slice(2);
+      correctFormat[i] = value;
+    } else if (correctFormat[i] === 'YYYY') {
+      if (Number(partsDate[fromFormat.indexOf('YY')]) < 30) {
+        value = `20${partsDate[fromFormat.indexOf('YY')]}`;
+      } else {
+        value = `19${partsDate[fromFormat.indexOf('YY')]}`;
+      }
+
+      correctFormat[i] = value;
+    }
```

Comment:
The logic for converting 'YY' to 'YYYY' and vice versa seems incorrect. The 'includes' check on line 30 is unnecessary because the formats 'YY' and 'YYYY' are mutually exclusive. Instead, you should check the length of the 'value' to determine if a conversion is needed. If the 'value' is 2 characters long, it's 'YY' and should be converted to 'YYYY' using the rules provided. If it's 4 characters long, and 'YY' is expected, then slice the last 2 characters.